### DISCUSSION TITLE
using notFound() gives Error: NEXT_NOT_FOUND

### DISCUSSION THREAD

**Discussion by simpel** [vercel/next.js#54896] [App Router]
Hello!

Whenever i try to use the notfound function in page.tsx it doesnt work and instead gives Error: NEXT_NOT_FOUND. Any ideas why this is happening? Is it not OK to use notFound() serverside? What am i missing?

For example this does not work

```typescript
const RootLayout = async ({ children, params }: TLayOut<TTemplate>) => {
	if (!params?.locale) notFound()

	const dictionary = await getDictionary(params.locale)

	return (
		<html lang={params ? params.locale : 'en'}>
			{children}		
		</html>
	)
}

export default RootLayout
``` 

My folder structure

![image](https://github.com/vercel/next.js/assets/232024/8e0e67c3-1146-48fd-92ab-96c93f1c2680)

**Comment by devjiwonchoi** [comment]
Assuming that you are calling `notFound` on the root layout, it is not allowed to be used on the root layout.

**Reply by simpel** [reply]
Ah! But how to handle errors etc inte the root layout then?

**Reply by devjiwonchoi** [reply]
Could do in various ways but quick answer: handle in `[locale]/layout.tsx`

**Reply by simpel** [reply]
I now removed everything from layout files so i only return notFound() in page.tsx. This works fine when running in dev mode, but the production build still gives:

```
Application error: a client-side exception has occurred (see the browser console for more information).
Digest: NEXT_NOT_FOUND
``` 

And there is noting logged in the console.

**Reply by devjiwonchoi** [reply]
Can you provide a repro? [CodeSandBox](https://codesandbox.io/) is handy.

**Reply by AprilNEA** [reply]
What is the reason for imposing restrictions? This greatly reduces flexibility.
